analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -18,58 +18,95 @@
-                && leftParameters.Value.Select((left, index) => left.IsEquivalentTo(rightParameters.Value[index])).All(x => x));
+                }
+            },
+            SyntaxKinds);
+
+    protected virtual bool IsExcludedFromBeingExamined(SonarSyntaxNodeReportingContext context) =>
+        context.ContainingSymbol.Kind != SymbolKind.NamedType;
+
+    protected static bool HaveSameParameters<TSyntax>(SeparatedSyntaxList<TSyntax>? leftParameters, SeparatedSyntaxList<TSyntax>? rightParameters)
+        where TSyntax : SyntaxNode =>
+        (leftParameters is null && rightParameters is null) // In VB.Net the parameter list can be omitted
+        || (leftParameters?.Count == rightParameters?.Count && HaveSameParameterLists(leftParameters.Value, rightParameters.Value));
+
+    protected static bool HaveSameTypeParameters<TSyntax>(SemanticModel model, SeparatedSyntaxList<TSyntax>? firstTypeParameterList, SeparatedSyntaxList<TSyntax>? secondTypeParameterList)
+        where TSyntax : SyntaxNode
+    {
+        var firstSymbols = firstTypeParameterList?.Select(x => model.GetDeclaredSymbol(x)).OfType<ITypeParameterSymbol>() ?? [];
+        var secondSymbols = secondTypeParameterList?.Select(x => model.GetDeclaredSymbol(x)).OfType<ITypeParameterSymbol>().ToArray() ?? [];
+        return firstSymbols.All(x => Array.Exists(secondSymbols, secondSymbol => TypesAreSame(x, secondSymbol)));
     }
+
+    private static bool HaveSameParameterLists<TSyntax>(SeparatedSyntaxList<TSyntax> leftParameters,
+                                                        SeparatedSyntaxList<TSyntax> rightParameters) where TSyntax : SyntaxNode =>
+        leftParameters.Equals(rightParameters, (left, right) => left.IsEquivalentTo(right)); // Perf: Syntactic equivalence for all parameters
+
+    private static bool TypesAreSame(ITypeSymbol leftParameterType, ITypeSymbol rightParameterType) =>
+        leftParameterType.Equals(rightParameterType) // M1(int x) <-> M2(int x)
+        || TypesAreSameTypeParameters(leftParameterType, rightParameterType) // M1<T>(T x) where T: class <-> M2<T>(T x) where T: class
+                                                                             // T of M1 is a different symbol than T of M2, but if they have the same constraints, they can be interchanged
+        || TypesAreSameGenericType(leftParameterType, rightParameterType); // M1<T>(IEnumerable<T> x) where T: class <-> M2<T>(IEnumerable<T> x) where T: class
+
+    private static bool TypesAreSameGenericType(ITypeSymbol leftParameterType, ITypeSymbol rightParameterType) =>
+        leftParameterType is INamedTypeSymbol { IsGenericType: true } namedTypeLeft
+        && rightParameterType is INamedTypeSymbol { IsGenericType: true } namedTypeRight
+        && namedTypeLeft.TypeArguments.Length == namedTypeRight.TypeArguments.Length
+        && namedTypeLeft.TypeArguments.Equals(namedTypeRight.TypeArguments, (x, y) =>
```

Comment:
This was changed from the last review:
* TypeArguments instead of TypeParameters
* for TypeArguments that are them self type parameters, name equality is enough (the constraints of the argument get checked in TypeParametersHaveSameNameAndConstraints. We can not recurse here because it will result in endless recursions. `<T1>() where T1: IEquatable<T1>` or `<T1, T2>() where T1: IEquatable<T2> where T2: IEquatable<T1>`